book/analysis/big-o-notation.md

```diff
@@ -75,15 +77,16 @@ of the problem. For these kinds of algorithms, we need to characterize their
 performances as *worst case*, *best case*, or *average case*.
 
 The worst case performance refers to a particular data set where the algorithm
-performs especially poorly, whereas a different data set might have amazing
-performance. In most cases, however, the algorithm performs somewhere in between
-these two extremes, in what we kindly call the average case. Understanding these
-distinctions can help prevent any one particular case from misleading us.
+performs especially poorly, while the best case performance, as you can
+probably infer, refers to a data set that the algorithm can process extremely
+fast. In most cases, however, the algorithm performs somewhere in between
```

Comment:
Isn't the mode what we're most interested in, though? I'm happy to remove the "most cases" bit, but for your example above, don't we care about the O(1) that occurs more frequently than we are the **true** average case? See what you think in my updated copy.